src/Elastic.Apm/Helpers/StacktraceHelper.cs

```diff
@@ -28,14 +28,15 @@ internal static List<CapturedStackFrame> GenerateApmStackTrace(StackFrame[] fram
 			try
 			{
 				retVal.AddRange(from item in frames
+					let fileName = item?.GetMethod()
+						?.ReflectedType?.FullName //see: https://github.com/elastic/apm-agent-dotnet/pull/240#discussion_r289619196
+					let functionName = GetRealMethodName(item?.GetMethod())
 					select new CapturedStackFrame
 					{
-						Function = GetRealMethodName(item?.GetMethod()),
-						FileName =
-							item?.GetMethod()
-								?.ReflectedType?.FullName, //see: https://github.com/elastic/apm-agent-dotnet/pull/240#discussion_r289619196
+						Function = functionName,
+						FileName = string.IsNullOrWhiteSpace(fileName)? "unknown" : fileName,
 						Module = item?.GetMethod()?.ReflectedType?.Assembly.FullName,
-						LineNo = item?.GetFileLineNumber() ?? 0
+						LineNo = item?.GetMethod().Name == functionName ? item?.GetFileLineNumber() ?? 0 : 0
```

Comment:
Doesn't compiler preserve original source code line numbers even after rewriting IL?

Reply:
I don't know, but I'll research this...

Reply:
Hahh, it does, but now the class name does not match - but we can also fix that... 

<img width="1202" alt="image" src="https://user-images.githubusercontent.com/1091853/59135324-d68d9b80-897e-11e9-8466-53e39c749a4b.png">

Reply:
@SergeyKleyman I think this makes sense to research more, it looks promising, but just to be on the safe side I'd rather not send the linenumber in case of async method for now. I think it'd be best to move on with this PR since it addresses a fairly important issue (all our events with stackframe are currently rejected by the server) and this change is just for being on the safe side and try to prevent sending incorrect info.

I opened https://github.com/elastic/apm-agent-dotnet/issues/266 to follow up on this.

Reply:
In the spirit of progress over perfection I would go with the opposite approach - sending line numbers for all the methods (async as well) and if we encounter an issue we will deal with it then. I think line number is very important information to just drop especially considering that we don't know of any case when it's incorrect.

I'm leaving it to you guys to decide.

Reply:
Ok, let's make that way.

Besides not being sure I just thought that it’s a little bit incorrect, since the class name we send does not match the method name, therefore the line number also does not match the class name (the `lineNo`  and the `function` seem to be correct, so that’s fine) - ok, I’ll change this and send line number, I don’t think sending it would be very bad, so let’s make it that way.

Also there is a followup issue, so we'll have another iteration on this question, let's send it always when we have it, and then go from there in #266.